remme/tp/node_account.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+from remme.protos.account_pb2 import AccountMethod
+from remme.protos.node_account_pb2 import (
+    NodeAccount, NodeAccountMethod, NodeAccountInternalTransferPayload, EmptyPayload
+)
+
+from remme.shared.forms import NodeAccountInternalTransferPayloadForm, EmptyPayloadForm
+from remme.shared.constants import Events, EMIT_EVENT
+
+from .basic import (
+    PB_CLASS, PROCESSOR, VALIDATOR, BasicHandler, get_data, get_multiple_data
+)
+from remme.settings.helper import _get_setting_value
+LOGGER = logging.getLogger(__name__)
+
+FAMILY_NAME = 'node_account'
+FAMILY_VERSIONS = ['0.1']
+
+
+def get_node_account_by_address(context, address):
+    account = get_data(context, NodeAccount, address)
+    if account is None:
+        return NodeAccount()
+    return account
+
+
+class NodeAccountHandler(BasicHandler):
+    def __init__(self):
+        super().__init__(FAMILY_NAME, FAMILY_VERSIONS)
+
+    def get_state_processor(self):
+        return {
+            NodeAccountMethod.INITIALIZE_MASTERNODE: {
+                PB_CLASS: NodeAccountInternalTransferPayload,
+                PROCESSOR: self._initialize_masternode,
+                VALIDATOR: NodeAccountInternalTransferPayloadForm,
+            },
+            NodeAccountMethod.CLOSE_MASTERNODE: {
+                PROCESSOR: self._close_masternode,
+                PB_CLASS: EmptyPayload,
+                VALIDATOR: EmptyPayloadForm,
+            },
+            NodeAccountMethod.TRANSFER_FROM_UNFROZEN_TO_OPERATIONAL: {
+                PB_CLASS: NodeAccountInternalTransferPayload,
+                PROCESSOR: self._transfer_from_unfrozen_to_operational,
+                VALIDATOR: NodeAccountInternalTransferPayloadForm,
+            }
+        }
+
+    def _initialize_masternode(self, context, node_account_public_key, internal_transfer_payload):
+        node_account_address = self.make_address_from_data(node_account_public_key)
+
+        node_account = get_node_account_by_address(context, node_account_address)
+
+        if node_account.node_state != NodeAccount.NodeState.Value('NEW'):
+            raise InvalidTransaction('Masternode is already opened or closed.')
+        minimum_stake = int(_get_setting_value(context, 'remme.settings.minimum_stake'))
+
+        if internal_transfer_payload.value < minimum_stake:
+            raise InvalidTransaction('Initial stake is too low.')
+
+        node_account.node_state = NodeAccount.NodeState.Value('OPENED')
+
+        node_account.balance -= internal_transfer_payload.value
+        frozen_part = minimum_stake
```

Comment:
You use `frozen_part` only once at 69th code line, why this variable is created?
If to clarify the business-logic, so leave a simple comment instead of to mess the code.

Reply:
Done. Variable removed.